File: src/main/java/net/atos/client/zgw/ztc/model/RoltypeListParameters.java
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.zgw.ztc.model;

import java.net.URI;

import jakarta.ws.rs.QueryParam;

import net.atos.client.zgw.ztc.model.generated.RolType;

/**
 *
 */
public class RoltypeListParameters extends AbstractZTCListParameters {

    /*
     * URL-referentie naar het ZAAKTYPE waar deze ROLTYPEn betrokken kunnen zijn.
     */
    @QueryParam("zaaktype")
    private URI zaaktype;

    /*
     * Algemeen gehanteerde omschrijving van de aard van de ROL.
     */
    private RolType.OmschrijvingGeneriekEnum omschrijvingGeneriek;

    public RoltypeListParameters(final URI zaaktype) {
        this.zaaktype = zaaktype;
    }

    public RoltypeListParameters(final RolType.OmschrijvingGeneriekEnum omschrijvingGeneriek) {
        this.omschrijvingGeneriek = omschrijvingGeneriek;
    }

    public RoltypeListParameters(final URI zaaktype, final RolType.OmschrijvingGeneriekEnum omschrijvingGeneriek) {
        this.zaaktype = zaaktype;
        this.omschrijvingGeneriek = omschrijvingGeneriek;
    }

    public URI getZaaktype() {
        return zaaktype;
    }

    @QueryParam("omschrijvingGeneriek")
    public String getOmschrijvingGeneriek() {
        return omschrijvingGeneriek == null ? null : omschrijvingGeneriek.value();
    }
}

File: src/main/kotlin/net/atos/client/zgw/ztc/model/RoltypeListParameters.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.ztc.model

import jakarta.ws.rs.QueryParam
import java.net.URI

class RoltypeListParameters(
    private val zaaktype: URI,
) : AbstractZtcListParameters() {
    /**
     * URL-referentie naar het ZAAKTYPE waar deze ROLTYPEn betrokken kunnen zijn.
     */
    @QueryParam("zaaktype")
    fun getZaaktype() = zaaktype
}
